reference/docs-conceptual/SecretManagement/how-to/using-secrets-in-automation.md

```diff
@@ -0,0 +1,84 @@
+---
+description: This article explains how to use a SecretStore vault in an automation scenario to securely retrieve an use passwords or other secret material.
+ms.date: 06/24/2022
+title: Using the SecretStore in automation
+---
+# Using the SecretStore in automation
+
+This article provides an example for using a **Microsoft.PowerShell.SecretStore** vault in an
+automation scenario. A **SecretStore** vault provides you a way to securely store and retrieve the
+passwords, tokens and other secrets you need to use in your automation pipeline.
+
+## Setting up the host the runs the automation
+
+For this example you must first install and configure the SecretManagement modules. This example
+assume that your automation host is running Windows. These commands must be run in the user context
+of the automation account on the host.
+
+```powershell
+Install-Module -Name Microsoft.PowerShell.SecretStore -Repository PSGallery -Force
+Install-Module -Name Microsoft.PowerShell.SecretManagement -Repository PSGallery -Force
+Import-Module Microsoft.PowerShell.SecretStore
+Import-Module Microsoft.PowerShell.SecretManagement
+```
+
+You must also create a password as a **SecureString** that is securely exported to an XML file and
+encrypted by Windows Data Protection (DPAPI). The following command prompts you for a password. In
+this example the **UserName** is unimportant.
+
+```powershell
+PS> $credential = Get-Credential -UserName 'SecureStore'
+
+PowerShell credential request
+Enter your credentials.
+Password for user SecureStore: **************
+```
+
+Once you have the password you can save it to an encrypted XML file.
+
+```powershell
+$securePasswordPath = 'C:\automation\passwd.xml'
+$credential.Password |  Export-Clixml -Path $securePasswordPath
+```
+
+Next you must configure the **SecretStore** vault. The configuration sets user interaction to
+`None`, so that **SecretStore** never prompts the user. The configuration requires a password,
+and the password is passed in as a **SecureString** object. The `-Confirm:false` parameter is used
+so that PowerShell does not prompt for confirmation.
+
+```powershell
+Register-SecretVault -Name SecretStore -ModuleName Microsoft.PowerShell.SecretStore -DefaultVault
+$password = Import-CliXml -Path $securePasswordPath
+
+$storeConfiguration = @{
+    Authentication = 'Password'
+    PasswordTimeout = 3600 # 1 hour
```

Comment:
The user may want to set the PasswordTimeout parameter to '-1' for automation, so that it remains in effect for the entire PowerShell session.  This way secrets are available to that session for the entire time it takes to run an automation script, which is usually unknown.  The idea is to run a single script or maybe a set of scripts that perform an important function, in a single PowerShell session and then close it down immediately afterwards.  The password can remain in effect for the entire session life since it exists only to run automation one time for a single purpose.

One potential drawback to setting an arbitrary password timeout is that it could timeout before the script finishes, causing the script to fail because it cannot access secrets.  Another strategy would be to retrieve all needed secrets just once and immediately in the script, so that the password is no longer needed as the script continues to run.

Reply:
I see that the PasswordTimeout parameter is discussed in more detail later.  So maybe we just need to mention that it is difficult to know how long for the password to remain in effect for automated script runs, and that allowing the password to exist for the entire session, and limiting that session to individual automation runs can be a good strategy.

Reply:
Yeah, this is discussed elsewhere. The purpose of this article is to show an example, not explain every detail.